Sample/Kitchen Sink/DemonstrationTools/UnitConverterViewModel.swift

```diff
@@ -0,0 +1,121 @@
+    @Published var outputUnit: LengthType = .metres
+    @Published var outputValue: String = ""
+    @Published var boundSelectionInput: LengthType? = .feet
+    @Published var boundSelectionOutput: LengthType? = .metres
+
+    private let feetToMetresConversionRate: Decimal = 3.28084
+    private let metresToFeetConversionRate: Decimal = 0.3048006096
+    private let kgToLbConversionRate: Decimal = 2.20462262
+
+    let adjustableConversionTitle = "Adjustable conversion"
+    let adjustableConversionSubTitle = "Example of conversion with Menu Picker"
+    let weightTitle = "Kilograms to Pounds"
+    let weightSubtitle = "Example of fixed weight conversion with reversable function"
+    let naivgationBarTitle = "Unit Converter"
+    let calculatedField = "Calculated result"
+    let convert = "Convert"
+
+    let lengthPalceholder = "Enter Length"
+    let bottomKgLabel = "Kilograms"
+    let bottomlbLabel = "Pounds"
+    let kgHint = "Enter Kgs"
+    let lbHint = "Enter Lgs"
+
+    func runLengthConversion() {
+        let inputDecimal = toDecimal(string: inputValue)
+        let inputUnitInMetres = convertToMeters(value: inputDecimal, from: boundSelectionInput ?? .metres )
+        let outputInMetres = convertFromMeters(value: inputUnitInMetres, from: boundSelectionOutput ?? .metres )
+        outputValue = toString2DP(value: outputInMetres)
+    }
+
+    func convertToMeters(value: Decimal, from unit: LengthType) -> Decimal {
+           switch unit {
+           case .feet:
+               return value / feetToMetresConversionRate
+           case .metres:
+               return value
+           }
+       }
+
+    func convertFromMeters(value: Decimal, from unit: LengthType) -> Decimal {
+        switch unit {
+        case .feet:
+            return value / metresToFeetConversionRate
+        case .metres:
+            return value
+        }
+    }
+
+     func toDecimal(string: String) -> Decimal {
+         return Decimal(string: string) ?? 0.0
+     }
+
+     func toString2DP(value: Decimal) -> String {
+         let formattedValue = getNSDecimalNumber(value: value)
+         return String(format: "%.2f", formattedValue.doubleValue)
+     }
+
+     func getNSDecimalNumber(value: Decimal) -> NSDecimalNumber {
+         let roundingHandler = NSDecimalNumberHandler(
+             roundingMode: .plain,
+             scale: 2,
+             raiseOnExactness: false,
+             raiseOnOverflow: false,
+             raiseOnUnderflow: false,
+             raiseOnDivideByZero: false
+         )
+         return NSDecimalNumber(decimal: value).rounding(accordingToBehavior: roundingHandler)
+     }
+
+    func convertKgsToLbs(kgs: Decimal) -> Decimal {
+        return (kgs * kgToLbConversionRate)
+    }
+
+    func convertLbsToKg(lbs: Decimal) -> Decimal {
+        return (lbs / kgToLbConversionRate)
+    }
+
+    func fieldsAreEmpty() -> Bool {
+        emptyFields = kgInputString.isEmpty && lbsInputString.isEmpty
```

Comment:
Minor/General: It's not ideal to have a function that sets a var and returns it, much cleaner to have the get built into the var e.g. which will avoid any source of truth issues too

```
var areFieldsEmpty: Bool {
    get {
        kgInputString.isEmpty && lbsInputString.isEmpty
    }
}
```